### PR TITLE
Allow to configure MAVEN_OPTS and JAVA_OPTS from codenvy.env

### PR DESCRIPTION
**Pull request by evoevodin**
### What does this PR do?
Allows to configure `MAVEN_OPTS` and `JAVA_OPTS` from codenvy.env.
Adaptation of eclipse/che#4119

#### Changelog
Allow to configure `MAVEN_OPTS` from codenvy.env

#### Release Notes
Allow to configure `MAVEN_OPTS` from codenvy.env. 
By default value provided by `CHE_WORKSPACE_JAVA_OPTIONS` is used.

#### Docs PR
N/A

### CODE REVIEW

**Comment by evoevodin** [comment]
@eivantsov or @JamesDrummond take a look at changelog & logs please

File: dockerfiles/init/manifests/codenvy.pp [hunk 1]
```diff
@@ -213,6 +213,8 @@
   $docker_registry_for_workspace_snapshots = getValue("CODENVY_DOCKER_REGISTRY_FOR_WORKSPACE_SNAPSHOTS","$host_url:5000")
   $workspace_auto_snapshot = getValue("CODENVY_WORKSPACE_AUTO_SNAPSHOT","false")
   $workspace_auto_restore = getValue("CODENVY_WORKSPACE_AUTO_RESTORE","false")
+  $workspace_java_options= getValue("CHE_WORKSPACE_JAVA_OPTIONS", "-Xms256m -Xmx2048m -Djava.security.egd=file:/dev/./urandom")
```
Comment (skabashnyuk) [inline dockerfiles/init/manifests/codenvy.pp]: Is it not too high? Xmx2048m? may be 1G?

Comment (evoevodin) [inline dockerfiles/init/manifests/codenvy.pp]: have no idea, just moved it from original property

File: dockerfiles/init/modules/codenvy/templates/machine.properties.erb [hunk 2]
```diff
@@ -86,7 +86,12 @@ che.docker.network_driver=<%= scope.lookupvar('codenvy::che_machine_docker_netwo
...
..., '').gsub(/^www./,'').split(':')[1].gsub(/\/.*/,'') %><% end -%><% end -%><% if ! @https_proxy_for_codenvy_workspaces.empty? -%><% if @https_proxy_for_codenvy_workspaces.include? '@' -%> -Dhttps.proxyUser=<%= @https_proxy_for_codenvy_workspaces.gsub(/^https?\:\/\//, '').gsub(/^www./,'').split('@')[0].split(':')[0] %> -Dhttps.proxyPassword=<%= @https_proxy_for_codenvy_workspaces.gsub(/^https?\:\/\//, '').gsub(/^www./,'').split('@')[0].split(':')[1] %> -Dhttps.proxyHost=<%= @https_proxy_for_codenvy_workspaces.gsub(/^https?\:\/\//, '').gsub(/^www./,'').split('@')[1].split(':')[0] %> -Dhttps.proxyPort=<%= @https_proxy_for_codenvy_workspaces.gsub(/^https?\:\/\//, '').gsub(/^www./,'').split('@')[1].split(':')[1].gsub(/\/.*/,'') %><% else -%> -Dhttps.proxyHost=<%= @https_proxy_for_codenvy_workspaces.gsub(/^https?\:\/\//, '').gsub(/^www./,'').split(':')[0] %> -Dhttps.proxyPort=<%= @https_proxy_for_codenvy_workspaces.gsub(/^https?\:\/\//, '').gsub(/^www./,'').split(':')[1].gsub(/\/.*/,'') %><% end -%><% end -%><% if ! @no_proxy_for_codenvy_workspaces.empty? -%> -Dhttp.nonProxyHosts='<%= @no_proxy_for_codenvy_workspaces.gsub(/^https?\:\/\//, '').gsub(/^www./,'').gsub(',','|') %>|'<% end -%>
```
Comment (riuvshin) [inline dockerfiles/init/modules/codenvy/templates/machine.properties.erb]: there should be space between `<%= scope.lookupvar('codenvy::workspace_java_options') %>`[:space:]`<% if ! @http_p...`

File: dockerfiles/init/manifests/codenvy.pp (hunk 1 above)
Comment (riuvshin) [inline dockerfiles/init/manifests/codenvy.pp]: with 1g it was failing with OOM, we increased to 2g a while ago

Comment (skabashnyuk) [inline dockerfiles/init/manifests/codenvy.pp]: 1g is failing to do what?

Comment (riuvshin) [inline dockerfiles/init/manifests/codenvy.pp]: to do java code assistant afaik, some times it was failing even without any codeasistant just by maven server

**Comment by JamesDrummond** [comment]
- Removed "By default value provided by `CHE_WORKSPACE_JAVA_OPTIONS` is used." from release notes.
- Added "and `JAVA_OPTS` workspace environment variables" to release and changelog. 

So this sets the workspace's environment variables? If workspace has  `MAVEN_OPTS` and `JAVA_OPTS` already set these values are overwritten by the agent? Do all workspace machine in multi-machine workspace have their environment variables set.

**Comment by slemeur** [comment]
Why did we kept the "CHE_WORKSPACE_JAVA_OPTIONS" with "CHE_" prefix here?

**Pull request merged**